### PR TITLE
Split vc-api and test-app into workspaces

### PR DESCRIPTION
**Pull request by nmattia**
This extracts the vc issuer's API into a new workspace and changes the test-app from a standalone package to a workspace of the main package.

By extracting the issuer API, the types (like credential request) can be reused elsewhere.

By turning the test-app into a workspace, we deduplicate a lot of the dependency information (by removing one package lock) and we can use local dependencies (like the issuer types).

### CODE REVIEW

File: src/vc-api/src/index.ts
Comment (nmattia) [inline src/vc-api/src/index.ts:1]: This file was just moved, not changed (beyond the extra export)

File: src/vc-api/README.md
Comment (peterpeterparker) [inline src/vc-api/README.md:10]: it's not `npm run build ...` instead of `npm run .... build`?

btw. you can shorten `--workspace` with `-w` if interesting

File: src/vc-api/package.json
Comment (peterpeterparker) [inline src/vc-api/package.json:10]: you need to redeclare those here given that you build from root?

btw. zod and dfinity libs are runtime libs no?

**Review by peterpeterparker — APPROVED** [review]
Already approving, if you can have a look at my two comments though before merging and if interesting.

File: src/vc-api/README.md
Comment (nmattia) [reply src/vc-api/README.md:10]: with `npm run build FOO`, it can be interpreted as "pass `FOO` to the build command", whereas with `npm run FOO build` it's clear that it's the build command that's being run inside the workspace

 > btw. you can shorten --workspace with -w if interesting
 
 I know :) but in scripts and READMEs I like the long form version for clarity

File: src/vc-api/package.json
Comment (nmattia) [reply src/vc-api/package.json:10]: oh yes that's right! good catch

File: src/vc-api/README.md
Comment (peterpeterparker) [reply src/vc-api/README.md:10]: > with npm run build FOO, it can be interpreted as "pass FOO to the build command", whereas with npm run FOO build it's clear that it's the build command that's being run inside the workspace

Clear is relative, I never seen `npm run -w ... build` before I have to say so it wasn't clear 😉

Comment (peterpeterparker) [reply src/vc-api/README.md:10]: Thanks for the explanation

**Comment by nmattia** [comment]
@peterpeterparker PTAL!

**Comment by peterpeterparker** [comment]
Already approved, go on!

**Pull request merged**
